sdk/digitaltwins/Azure.DigitalTwins.Core/src/Customized/DigitalTwinsRestClient.cs

```diff
@@ -804,38 +872,49 @@ private HttpMessage CreateUpdateComponentRequest(string id, string componentPath
+        private HttpMessage CreateSendTelemetryRequest(string id, string dtId, string telemetry, string dtTimestamp)
         {
-            HttpMessage message = _pipeline.CreateMessage();
-            Request request = message.Request;
+            var message = _pipeline.CreateMessage();
+            var request = message.Request;
             request.Method = RequestMethod.Post;
             var uri = new RawRequestUriBuilder();
             uri.Reset(endpoint);
             uri.AppendPath("/digitaltwins/", false);
             uri.AppendPath(id, true);
             uri.AppendPath("/telemetry", false);
+            uri.AppendQuery("api-version", apiVersion, true);
             request.Uri = uri;
-            request.Headers.Add("ce-id", ceId);
-            if (ceDatacontenttype != null)
-            {
-                request.Headers.Add("ce-datacontenttype", ceDatacontenttype);
-            }
-            request.Headers.Add("ce-dataschema", ceDataschema);
-            request.Headers.Add("ce-specversion", ceSpecversion);
-            if (ceSubject != null)
+            request.Headers.Add("dt-id", dtId);
+            if (dtTimestamp != null)
             {
-                request.Headers.Add("ce-subject", ceSubject);
+                request.Headers.Add("dt-timestamp", dtTimestamp);
             }
-            if (ceTime != null)
-            {
-                request.Headers.Add("ce-time", ceTime);
-            }
-            request.Headers.Add("ce-type", ceType);
-            request.Headers.Add("Content-Type", "application/json; charset=utf-8");
-            if (telemetry != null)
+            request.Headers.Add("Content-Type", "application/json");
+            request.Content = new StringRequestContent(telemetry);
+            return message;
+        }
+
+        private HttpMessage CreateSendComponentTelemetryRequest(string id, string componentPath, string dtId, string telemetry, string dtTimestamp)
+        {
+            var message = _pipeline.CreateMessage();
+            var request = message.Request;
+            request.Method = RequestMethod.Post;
+            var uri = new RawRequestUriBuilder();
+            uri.Reset(endpoint);
+            uri.AppendPath("/digitaltwins/", false);
+            uri.AppendPath(id, true);
+            uri.AppendPath("/components/", false);
+            uri.AppendPath(componentPath, true);
+            uri.AppendPath("/telemetry", false);
+            uri.AppendQuery("api-version", apiVersion, true);
+            request.Uri = uri;
+            request.Headers.Add("dt-id", dtId);
+            if (dtTimestamp != null)
             {
-                request.Content = new StringRequestContent(telemetry);
+                request.Headers.Add("dt-timestamp", dtTimestamp);
             }
+            request.Headers.Add("Content-Type", "application/json");
+            request.Content = new StringRequestContent(telemetry);
```

Comment:
What if telemetry is null?

Reply:
There is a guard against telemetry being null in the calling method. But to be on the safe side, I will guard against null here as well.